### ISSUE TITLE
An error occurred when 00.train generated input.json

### ISSUE THREAD

**Issue by caojiachun** [deepmodeling/dpgen#1286]
The `input.json` generated by the training procedure type does not write `training_data` properly:

![image](https://github.com/deepmodeling/dpgen/assets/71744229/788c9f21-8a53-406a-a8e5-00557500cccf)

*/iter.000000/00.train/000/input.json:*
    ...
    "training": {
        "stop_batch": 300000,
        "disp_file": "lcurve.out",
        "disp_freq": 1000,
        "numb_test": 1,
        "save_freq": 5000,
        "save_ckpt": "model.ckpt",
        "disp_training": true,
        "time_training": true,
        "profiling": false,
        "profiling_file": "timeline.json",
        "_comment": "that's all",
        "training_data": {
            `"systems": [],`
            `"batch_size": []`
        },
        "seed": 482819826
    }

*my param.json:*
{
    "type_map": [
        "H",
        "C",
        "N",
        "O",
        "S",
        "Fe"
    ],
    "mass_map": [
        1.00784,
        12.0107,
        14.0067,
        15.9994,
        32.065,
        55.845
    ],
    "init_data_prefix": "./data",
    "init_data_sys": [
        "/vac_FeN4_PMS/deepmd/H1C90N4O5S1Fe1/"
    ],
    "init_batch_size": ["auto"],
    "sys_configs_prefix": "./",
    "sys_configs": [
                ["init_lammps/vac_FeN4_pms/0*/POSCAR"]
    ],
    "_comment": " that's all ",
    "ratio_failed": 1,
    "numb_models": 4,
    "default_training_param": {
        "model": {
            "type_map": [ 
                "H",
                "C",
                "N",
                "O",
                "S",
                "Fe"
                        ],
            "descriptor": {
                "type": "se_a",
                "sel": [1, 39, 4, 5, 1, 1],
                "rcut_smth": 1.0,
                "rcut": 6.0,
                "neuron": [
                    20,
                    40,
                    80
                ],
                "resnet_dt": false,
                "axis_neuron": 12,
                "seed": 1
            },
            "fitting_net": {
                "neuron": [
                    240,
                    240,
                    240
                ],
                "resnet_dt": true,
                "seed": 1
            }
        },
        "learning_rate": {
            "type": "exp",
            "start_lr": 0.001,
            "stop_lr":  1e-8,
            "decay_steps": 1000
        },
        "loss": {
            "start_pref_e": 0.02,
            "limit_pref_e": 0.1,
            "start_pref_f": 1000,
            "limit_pref_f": 1,
            "start_pref_v": 0.0,
            "limit_pref_v": 0.0
        },
        "training": {
            "stop_batch": 300000,
            "disp_file": "lcurve.out",
            "disp_freq": 1000,
            "numb_test": 1,
            "save_freq": 5000,
            "save_ckpt": "model.ckpt",
            "disp_training": true,
            "time_training": true,
            "profiling": false,
            "profiling_file": "timeline.json",
            "_comment": "that's all"
        }
    },
    "model_devi_dt": 0.0003,
    "model_devi_plumed": false,
    "model_devi_skip": 0,
    "model_devi_f_trust_lo": 0.3,
    "model_devi_f_trust_hi": 0.4,
    "_model_devi_e_trust_lo": 10000000000.0,
    "_model_devi_e_trust_hi": 10000000000.0,
    "model_devi_clean_traj": true,
    "model_devi_jobs": [
                {
                                "sys_idx": [0],"temps": [300,400], "press": [1.0], "trj_freq": 20, "nsteps": 8000,"ensemble": "nvt", "_idx": "00"
                }
    ],
    "fp_style": "cp2k",
    "shuffle_poscar": false,
    "fp_task_max": 300,
    "fp_task_min": 50,
    "fp_pp_path": "./",
    "fp_pp_files": [],
    "external_input_path": "./cp2k.inp"
}

**Comment by njzjz** [comment]
It seems that you provide an absolute path in `init_data_sys`.

**Comment by caojiachun** [comment]
> It seems that you provided an absolute path in `init_data_sys`.

Thanks for your suggestion, it works fine after changing to absolute path.

    "init_data_prefix": "./",
    "init_data_sys": [
        "data/vac_FeN4_PMS/deepmd/H1C90N4O5S1Fe1/"
    ],
